xfp/tests/xiter/test_xiter_methods.py

```diff
@@ -151,6 +198,29 @@ def test_xiter_tail_fx_fail():
     assert isinstance(actual.value, IndexError) and actual.branch == XRBranch.LEFT
 
 
+def test_xiter_takewhile():
+    input = Xiter(itertools.count(0, 1))
+    actual = input.takewhile(lambda x: x < 10)
+    expected = Xiter(range(10))
+    assert compare(actual, expected)
+    assert next(input) == 0
+
+
+def test_xiter_takeuntil():
+    input = Xiter(itertools.count(0, 1))
+    actual = input.takeuntil(lambda x: x >= 10)
+    expected = Xiter(range(10))
+    assert compare(actual, expected)
+    assert next(input) == 0
+
+
+def test_xiter_to_xlist():
+    input = Xiter(range(5))
+    assert input.to_Xlist() == Xlist([0, 1, 2, 3, 4])
+    with pytest.raises(StopIteration):
+        next(input) == 0
```

Comment:
to_Xlist consomme l'itérateur ?

Reply:
Je confirme, c'est discutable.
Le principe appliqué, toutes les méthodes équivalentes à du slice ( slice / take / takewhile / takeuntil ...) ne consomme pas.

Ici on construit une Xlist. Par défaut Xlist récupère l'iterator depuis  l'itérable en argument, sans copie
-> dans le cas d'un iterator, on consomme celui-ci sans le copier, on le consomme

Si on veut changer le comportement, je pense que c'est au niveau du constructeur Xlist qu'il faut un cas spécial pour les iterators ou l'on passe par la copie. Je te laisse me dire ce que tu en pense.

Reply:
mmmh pas sûr, étant donné qu'il y a une notion de propriétaire de la donnée qui rentre en jeu : 
- construction de xlist : je ne possède pas l'itérable donc je laisse le soin à l'utilisateur de copier ou non
- to_xlist : je possède l'itérable, donc je le copie avant de le transformer de n'importe quelle façon que ce soit

effectivement, à rediscuter, il y a peut-être d'autres paramètres qui rentrent en jeu

Reply:
tu m'as convaincu